UtilitiesProject/Helpers.cs

```diff
@@ -1,10 +1,199 @@
+            try
+            {
+                int propertyNamePosition = 0, operationPosition = 1, filterPosition = 2;
+                if (filter is IList<object> filterList)
+                {
+                    //it is checking all way.
+                    if (filterList[propertyNamePosition].GetType().Name == typeof(string).Name)
+                    {
+                        //single filter
+                        query = SingleLogic(query, filterList, propertyNamePosition, operationPosition, filterPosition).AsQueryable();
+                    }
+                    else if (filterList[propertyNamePosition].GetType().Name == typeof(JArray).Name)
+                    {
+                        //it is checking multiple filter.
+                        int row = 0;
+                        foreach (var item in filterList.ToList())
+                        {
+                            if (item.GetType().Name == typeof(string).Name)
+                            {
+                                //single filter with and/or/strings
+                                query = SingleLogic(query, filterList, propertyNamePosition, operationPosition, filterPosition, (row == operationPosition)).AsQueryable();
+                            }
+                            else if (item.GetType().Name == typeof(JArray).Name)
+                            {
+                                if (item is JArray jArray)
+                                {
+                                    List<object> itemlists = jArray.ToObject<List<object>>();
+                                    query = SingleLogic(query, itemlists, propertyNamePosition, operationPosition, filterPosition).AsQueryable();
+
+                                }
+                            }
+                            row++;
+                        }
+                    }
+                    //}
+                    result = query;
+                }
+
+
+                result = query;
+
+
+            }
+            catch (Exception ex)
+            {
+
+                return query;
+            }
+            // If no filters apply, return the full list
+            return result;
+        }
+
+        /// <summary>
+        /// Logic to Express query with filter multiple as well as single
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="query"></param>
+        /// <param name="filterList"></param>
+        /// <param name="propertyNamePosition"></param>
+        /// <param name="operationPosition"></param>
+        /// <param name="filterPosition"></param>
+        /// <param name="oprationKeyWord"></param>
+        /// <returns></returns>
+        /// <exception cref="NotSupportedException"></exception>
+        internal static IQueryable<T> SingleLogic<T>(IQueryable<T> query, IList<object> filterList, int propertyNamePosition, int operationPosition, int filterPosition, bool oprationKeyWord = false)
```

Comment:
Rename with ApplyFilter

Reply:
Actually that fucntion is calling **ApplyFilter**.

Reply:
try "PrepareFilterQuery"